cylc/flow/scheduler.py

```diff
@@ -1138,7 +1138,7 @@ def remove_tasks(
                 # Check if downstream task should remain spawned:
                 if (
                     # Ignoring tasks we are already dealing with:
-                    child_itask.identity in matched_task_ids
+                    child_itask.tokens.task in matched_tasks
```

Comment:
Not sure why Mypy didn't catch this non-overlapping comparison (`if str in set[Tokens]`) 🤔